helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -459,14 +520,21 @@ public void updateDeploymentGroupHosts(final String groupName, final List<String
         final List<ZooKeeperOperation> ops = Lists.newArrayList();
         ops.add(set(Paths.statusDeploymentGroupHosts(groupName), Json.asBytes(hosts)));
 
-        final DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
+        DeploymentGroup deploymentGroup = getDeploymentGroup(groupName);
         ImmutableList<Map<String, Object>> events = ImmutableList.of();
 
         if (deploymentGroup.getJobId() != null) {
-          final DeploymentGroupStatus deploymentGroupStatus = getDeploymentGroupStatus(groupName);
-          if (deploymentGroupStatus == null || deploymentGroupStatus.getState() != FAILED) {
-            final RollingUpdateOp op =
-                getInitRollingUpdateOps(deploymentGroup, hosts, HOSTS_CHANGED, client);
+          if (updateOnHostChange(deploymentGroup, status)) {
```

Comment:
Good point. I've taken a pass in https://github.com/spotify/helios/pull/932/commits/425a04f9e3749545af2d1a4281d2db0613d94a68. Does that look about right?

I feel like `updateDeploymentGroupHosts` is getting a bit bloated, so I'm open to refactoring if anyone has any ideas.